### PR TITLE
Add plugin definition 

### PR DESCRIPTION
**Pull request by Seaven**
#2345

### CODE REVIEW

File: be/src/plugin/plugin.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace doris {
+
+#define DORIS_PLUGIN_VERSION 0x0001
+
+typedef struct st_plugin {
```
Comment (imay) [inline be/src/plugin/plugin.h]: Please give more comment for this class and its field, better to add some example for user. Make them easy to add one plugin.

File: be/src/plugin/plugin_manager.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "string"
+#include "unordered_map"
```
Comment (imay) [inline be/src/plugin/plugin_manager.h]: for system include, use `<>`

File: be/src/plugin/plugin_manager.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "common/status.h"
+#include "plugin.h"
```
Comment (imay) [inline be/src/plugin/plugin_manager.h]: use full path for inlcude "plugin/plugin.h"

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/plugin/plugin.h (hunk 1 above)
Comment (Seaven) [reply be/src/plugin/plugin.h]: OK, will modify and will support example when the feature finish

File: fe/src/main/java/org/apache/doris/plugin/PluginType.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.plugin;
+
+public enum PluginType {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/plugin/PluginType.java]: Add comments

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**